src/operators/Remap/avtRemapFilter.C

```diff
@@ -221,75 +229,42 @@ avtRemapFilter::Execute(void)
     // ------------------------------------ //
 
     avtDataTree_p inTree = GetInputDataTree();
-    TraverseDomainTree(inTree);
-    
-    SetOutputDataTree(new avtDataTree(rg, 0));
-    debug5 << "DONE Remapping" << std::endl;
-
-    CleanClippingFunctions();
-    return;
-}
-
-
-// ****************************************************************************
-//  Method: avtRemapFilter::TraverseDomainTree
-//
-//  Purpose:
-//      Traverse the DomainTree.
-//
-//  Arguments:
-//      inTree     avtDataTree_p The root of the domain tree from Execute or
-//                 else a node in the domain tree called during recursion.
-//
-//  Programmer: rusu1
-//  Creation:   Wed Apr  3 13:52:34 PDT 2019
-//
-// ****************************************************************************
-
-void
-avtRemapFilter::TraverseDomainTree(avtDataTree_p inTree)
-{
-    debug3 << "avtRemapFilter::TraverseDomainTree" << std::endl;
-    if (*inTree == NULL)
+    int totalNodes;
+    vtkDataSet **dataSets = inTree->GetAllLeaves(totalNodes);
+    for (int i = 0; i < totalNodes; ++i)
     {
-        debug4 << "inTree is null" << std::endl;
-        return;
+        ClipDomain(dataSets[i]);
     }
-    
-    std::vector<int> domainIds;
-    inTree->GetAllDomainIds(domainIds);
-    
-    int numChildren = inTree->GetNChildren();
-    
-    if (numChildren <= 0 && !inTree->HasData())
+
+#ifdef PARALLEL
+    int size = vars->GetNumberOfTuples();
+    double *varsDouble = (double*) vars->GetVoidPointer(0);
+    double *newBuff = new double[size];
+    // SumDoubleArrayAcrossAllProcessors(varsDouble, newBuff, size);
+    SumDoubleArray(varsDouble, newBuff, size);
+    if (PAR_Rank() == 0)
     {
-        debug4 << "No children and no data" << std::endl;
-        return;
+        for (int i = 0; i < size; ++i)
+        {
+            varsDouble[i] = newBuff[i];
+        }
     }
-    
-    if (numChildren == 0)
+    delete [] newBuff;
+#endif
+
+    if (PAR_Rank() == 0)
     {
-        debug5 << "Number of children is 0. Clipping this domain." << std::endl;
-        debug5 << "Domain Id: " << domainIds[0] << std::endl;
-        ClipDomain(inTree);
-        return;
+        SetOutputDataTree(new avtDataTree(rg, 0));
     }
     else
     {
-        debug5 << "Number of children is " << numChildren
-               << ". Looping over children." << std::endl;
-        for (int i = 0; i < numChildren; ++i)
-        {
-            if (inTree->ChildIsPresent(i))
-            {
-                TraverseDomainTree(inTree->GetChild(i));
-            }
-            else
-            {
-                debug4 << "Child " << i << " is not present. Skipping." << std::endl;
-            }
-        }
+        rg->Delete();
```

Comment:
I think you should remove the "rg->Delete()" line. My assumption was that you created rg in this method, so you needed to clean it up if you weren't returning it. It turns out you create it in the constructor and delete it in the destructor.

Reply:
Re-reading your comment, you already tried that. I'm sure you don't want the line "rg->Delete()", so there must still be something else going wrong.